platform/jobs/RetirementJobs.groovy

```diff
@@ -0,0 +1,222 @@
+/* UserRetirementJobs.groovy
+ *
+ * Defines jobs which orchestrate user retirement.
+ */
+
+package platform
+
+import static org.edx.jenkins.dsl.JenkinsPublicConstants.JENKINS_PUBLIC_TEAM_SECURITY
+import static org.edx.jenkins.dsl.JenkinsPublicConstants.JENKINS_PUBLIC_LOG_ROTATOR
+
+
+// ########### user-retirement-driver ###########
+// This defines the job for retiring individual users.
+job('user-retirement-driver') {
+    description('Drive the retirement of a single user which is ready for retirement immediately.')
+
+    // Only a subset of edx employees should be allowed to control this job,
+    // but customer support can read and discover.
+    authorization {
+        blocksInheritance(true)
+        List membersWithFullControl = ['edx/platform-team']
+        membersWithFullControl.each { emp ->
+            permissionAll(emp)
+        }
+        // TODO PLAT-2036: uncomment the following two lines when we add the
+        // appropriate github group.
+        //permission('hudson.model.Item.Read', 'edx/customer-support')
+        //permission('hudson.model.Item.Discover', 'edx/customer-support')
+    }
+
+    // The jenkins-worker is intended for platform workers, but they'll work
+    // well for this job, at least for now.  Specifically, this label is
+    // configured to disallow more than one simultaneous job per instance, and
+    // generally there are always available jenkins-worker instances idling.
+    label('jenkins-worker')
+
+    // Allow this job to have simultaneous instances building concurrently.
+    // This will make it possible for retirement of multiple users to progress
+    // in parallel.
+    concurrentBuild(true)
+
+    // keep jobs around for 30 days
+    logRotator JENKINS_PUBLIC_LOG_ROTATOR(30)
+
+    wrappers {
+        buildUserVars() /* gives us access to BUILD_USER_ID, among other things */
+        buildName('#${BUILD_NUMBER}, ${ENV,var="ORIGINAL_USERNAME"}')
+        timestamps()
+        colorizeOutput('xterm')
+    }
+
+    parameters {
+        stringParam('TUBULAR_BRANCH', 'master', 'Repo branch for the tubular scripts.')
+        stringParam('ORIGINAL_USERNAME', '', '')
+        stringParam('RETIRED_USERNAME', '', '')
+        stringParam('ORIGINAL_EMAIL', '', '')
+        stringParam('RETIRED_EMAIL', '', '')
+        stringParam('CURRENT_STATE', 'PENDING', 'Current retirement state for this user before starting this build.')
+        stringParam('LAST_STATE', '', 'Last retirement state for this user before starting this build.')
```

Comment:
@doctoryes except for TUBULAR_BRANCH, these are the names of the keys that should be present in each parameter file from the master python script.  Whether or not these are actually right as-written, this list needs to be be in sync with whatever the parameter files contain.